packages/visx-text/src/Text.tsx

```diff
@@ -17,216 +15,39 @@ function isValidXOrY(xOrY: string | number | undefined) {
   );
 }
 
-interface WordWithWidth {
-  word: string;
-  width: number;
-}
-
-interface WordsWithWidth {
-  words: string[];
-  width?: number;
-}
-
-type SVGTSpanProps = React.SVGAttributes<SVGTSpanElement>;
-type SVGTextProps = React.SVGAttributes<SVGTextElement>;
-
-type OwnProps = {
-  /** className to apply to the SVGText element. */
-  className?: string;
-  /** Whether to scale the fontSize to accommodate the specified width.  */
-  scaleToFit?: boolean;
-  /** Rotational angle of the text. */
-  angle?: number;
-  /** Horizontal text anchor. */
-  textAnchor?: 'start' | 'middle' | 'end' | 'inherit';
-  /** Vertical text anchor. */
-  verticalAnchor?: 'start' | 'middle' | 'end';
-  /** Styles to be applied to the text (and used in computation of its size). */
-  style?: React.CSSProperties;
-  /** Ref passed to the Text SVG element. */
-  innerRef?: React.Ref<SVGSVGElement>;
-  /** x position of the text. */
-  x?: string | number;
-  /** y position of the text. */
-  y?: string | number;
-  /** dx offset of the text. */
-  dx?: string | number;
-  /** dy offset of the text. */
-  dy?: string | number;
-  /** Desired "line height" of the text, implemented as y offsets. */
-  lineHeight?: SVGTSpanProps['dy'];
-  /** Cap height of the text. */
-  capHeight?: SVGTSpanProps['capHeight'];
-  /** Font size of text. */
-  fontSize?: string | number;
-  /** Font family of text. */
-  fontFamily?: string;
-  /** Fill color of text. */
-  fill?: string;
-  /** Maximum width to occupy (approximate as words are not split). */
-  width?: number;
-  /** String (or number coercible to one) to be styled and positioned. */
-  children?: string | number;
-};
-
-export type TextProps = OwnProps & Omit<SVGTextProps, keyof OwnProps>;
-
-type TextState = {
-  wordsByLines: WordsWithWidth[];
-};
-
-class Text extends React.Component<TextProps, TextState> {
-  static defaultProps = {
-    x: 0,
-    y: 0,
-    dx: 0,
-    dy: 0,
-    lineHeight: '1em',
-    capHeight: '0.71em', // Magic number from d3
-    scaleToFit: false,
-    textAnchor: 'start',
-    verticalAnchor: 'end', // default SVG behavior
-  };
-
-  state: TextState = {
-    wordsByLines: [],
-  };
-
-  private wordsWithWidth: WordWithWidth[] = [];
-
-  private spaceWidth: number = 0;
-
-  componentDidMount() {
-    this.updateWordsByLines(this.props, true);
+export default function Text(props: TextProps) {
+  const {
+    dx = 0,
+    dy = 0,
+    textAnchor = 'start',
+    innerRef,
+    verticalAnchor,
+    angle,
+    lineHeight = '1em',
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
this shouldn't be needed (typically if there is a `...rest`, the linter is smart enough not to flag)

Reply:
The reason why I put `scaleToFit` out of `textProps` is that `sacle-to-fit` isn't a legal SVG attribute
https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute

Otherwise `scaleToFit` wrapped in `textProps`, browser will throw an error.

Reply:
Actually It can't :joy: :joy: :joy:
![Screen Shot 2020-12-04 at 09 23 09](https://user-images.githubusercontent.com/20318608/101109799-aa55e800-3612-11eb-83db-080c6249742f.png)
![Screen Shot 2020-12-04 at 09 24 15](https://user-images.githubusercontent.com/20318608/101109805-ac1fab80-3612-11eb-9372-ca933a770891.png)

Reply:
oh no! 🙉  sorry for the back and forth.